SwiftSampleProject/SwiftyVokoderTests/ManagedObjectContextTests.swift

```diff
@@ -0,0 +1,160 @@
+//
+//  ManagedObjectContextTests.swift
+//  SwiftSampleProject
+//
+//  Created by Carl Hill-Popper on 2/9/16.
+//  Copyright © 2016 Vokal.
+//
+
+import XCTest
+import Vokoder
+@testable import SwiftyVokoder
+
+class ManagedObjectContextTests: XCTestCase {
+
+    let manager = VOKCoreDataManager.sharedInstance()
+    
+    override func setUp() {
+        super.setUp()
+        self.manager.resetCoreData()
+        self.manager.setResource("CoreDataModel", database: "CoreDataModel.sqlite")
+        
+        Stop.vok_import(CTAData.allStopDictionaries())
+        self.manager.saveMainContextAndWait()
+    }
+    
+    func testContextChain() {
+        let tempContext = self.manager.temporaryContext()
+        //temp context is a child of the main context
+        XCTAssertEqual(tempContext.parentContext, self.manager.managedObjectContext)
+        //main context has a private parent context
+        XCTAssertNotNil(self.manager.managedObjectContext.parentContext)
+    }
+    
+    func testDeletingObjectsOnTempContextGetsSavedToMainContext() {
+        //get a temp context, delete from temp, save to main, verify deleted on main
+        
+        let tempContext = self.manager.temporaryContext()
+        
+        let countOfStations = self.manager.countForClass(Station.self)
+        XCTAssert(countOfStations > 0)
+        
+        tempContext.performBlockAndWait {
+            self.manager.deleteAllObjectsOfClass(Station.self, context: tempContext)
+            self.manager.saveAndMergeWithMainContextAndWait(tempContext)
```

Comment:
Got it. But that means your call to `saveAndMergeWithMainContextAndWait` here doesn't need to happen within the block, since that method itself winds up running in a `performBlock` call anyway, right?